### PR TITLE
Gate the test-realm SW so it stops intercepting when unclaimed

### PR DESCRIPTION
**Pull request by lukemelia**
## Problem

Host test shards fail intermittently with whole modules failing at setup (`Promise rejected before … [object Object]`) or hanging to a 60s timeout, with the app stuck on the **login form**. Recent examples on #5373 hit `theme-card-test`, `fallback-models`, `show-card`, `workspace-chooser-delete`, and `workspace-chooser archive`.

## Root cause: the test-realm service worker leaks across modules

`packages/host/public/test-realm-sw.js` intercepts every browser-level fetch to `http://test-realm/…` and relays it to the window via `postMessage`, returning `503 "No responsive client available"` if nothing replies within its **1500ms** relay timeout.

The client-side responder (the `test-realm-fetch` handler in `tests/helpers/test-realm-service-worker.ts`, via `setupTestRealmServiceWorker`) is installed **only** by the ~13 `image-def/*` and `audio-def/*` acceptance modules.

`unregister()` (called in that helper's `hooks.after`) does **not** evict an already-active worker from a still-loaded page — the active worker keeps controlling the QUnit runner until the page unloads. So once an image/audio module registers the SW, it lingers into later modules. A later module that does a `http://test-realm/` browser fetch but never installed a responder — e.g. boot-time `MatrixService.start → setSystemCard → store.get(http://test-realm/test/SystemCard/default)` — gets a **503**. Because a 503 (unlike the expected **404** for a missing SystemCard) is fatal to boot, the app renders the login form, and the module cascades or times out (60s waiting for operator-mode DOM that never appears).

Confirmed from the failing logs:
- The 503 body `No responsive client available` is emitted only at `test-realm-sw.js:~29`.
- Timeout diagnostics show the app fully settled and stuck on the login form; the boot failure is the SystemCard 503 during `IndexRootRoute.model`.
- Module order in a failing shard: `Acceptance | image def | opening an image URL in a new tab` (registers SW) → `Acceptance | workspace-chooser-delete` (victim), i.e. the exact leaked-SW adjacency.
- The `relation "jobs" does not exist` / `worker-fatal` lines that also appear are **shutdown-time noise** (timestamped after tests finish); not causal.

**Why it's flaky / shard-composition-sensitive:** it only bites when an image/audio module lands immediately before a non-SW module that fetches `test-realm/` URLs in the same shard/browser session. Adding tests reshuffles shard boundaries, so a PR that adds tests (or a stale branch) can trigger it while `main` doesn't.

## Fix

Gate interception on an `active` flag in the SW:
- Defaults to **passthrough** (off).
- The owning module turns it **on** in `beforeEach` (after the responder is listening) and **off** in `hooks.after` (before `unregister`), each via a small ack'd `postMessage`.
- A lingering worker then passes `test-realm/` requests straight through — behaving exactly as if no SW were installed, which is the state non-intercepting modules already expect and pass under. No responder is invoked for them, so there is no hanging-fetch failure mode.
- The flag defaults off so a terminated-and-restarted worker (which loses the in-memory flag) fails safe toward passthrough rather than resurrecting the leak; the owning module re-asserts it every `beforeEach`.

## Rejected alternative

An earlier attempt (on #5373, since reverted) made the responder **always-on** — installed globally and resolving the network at message time. That backfired: it called `virtualNetwork.fetch()` for every leaked request, and for slow/not-found/torn-down fetches the promise never resolved, holding a test waiter so `settled()` never completed — turning fast cascades into 243s timeouts and spreading failures from one shard to three. This PR takes the opposite tack (stop intercepting) rather than (always respond).

## Validation

Host tests can't be run locally (the `ember test --path dist` harness aborts at boot), so this is validated by CI on this branch. Lint + type-check pass locally.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by lukemelia** [comment]
[Claude Code 🤖] CI green: all 20 Host Test shards pass, including 10/12/13 which were the leaked-SW victims. Marking ready for review.

File: packages/host/tests/helpers/test-realm-service-worker.ts [hunk 1]
```diff
@@ -36,6 +36,36 @@ async function ensureRegistered(): Promise<void> {
...
+      channel.port1.onmessage = null;
+      resolve();
+    };
+    let timeout = setTimeout(finish, 500);
+    channel.port1.onmessage = finish;
+    controller.postMessage({ type: 'test-realm-sw-set-active', active }, [
+      channel.port2,
+    ]);
+  });
+}
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/host/tests/helpers/test-realm-service-worker.ts:74]: [Claude Code 🤖] Split this into the two concerns:

**Ports:** good catch — fixed in 6a4901d. `finish()` now closes `port1` (and the SW closes the transferred ack port on its side), mirroring `relayViaClient`'s cleanup.

**Wrong controller:** leaving this as-is. This helper runs only in tests, and the SW's `fetch` handler only fires when test-realm-sw is the *controlling* worker — the entire relay already depends on that invariant, independent of this flag. If some other SW were controlling, the image/audio modules would fail regardless (their fetches would never reach the responder), so CI passing those modules confirms test-realm-sw is the controller when `setSwActive` runs. Posting to `swRegistration.active` instead wouldn't help either: if test-realm-sw isn't the controller, setting its in-memory flag is a no-op anyway.

File: packages/host/public/test-realm-sw.js [hunk 2]
```diff
@@ -6,7 +6,36 @@ self.addEventListener('activate', (event) => {
...
+self.addEventListener('message', (event) => {
+  let data = event.data;
+  if (!data || data.type !== 'test-realm-sw-set-active') {
+    return;
+  }
+  active = Boolean(data.active);
+  let port = event.ports && event.ports[0];
+  if (port) {
+    port.postMessage({ ok: true, active });
+  }
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/host/public/test-realm-sw.js:38]: [Claude Code 🤖] Fixed in 6a4901d — the handler now calls `port.close()` after posting the ack, matching `relayViaClient`'s port cleanup.

**Review by habdelra — APPROVED** [review]
